core/src/env/srml/srml_only/impls.rs

```diff
@@ -94,7 +95,7 @@ where
 impl<T> Env for SrmlEnv<T>
 where
     T: EnvTypes,
-    <T as EnvTypes>::Address: for<'a> TryFrom<&'a [u8]>,
+    <T as EnvTypes>::Address: Decode,
     <T as EnvTypes>::Hash: for<'a> TryFrom<&'a [u8]>,
```

Comment:
Those two lines look asymmetric. Would have suspected that `Hash` requires the same interface now. Also I am not sure if we are still in need of `EnvTypes` due to `node-runtime`.